### PR TITLE
Replace providerId/consumerId with odrl:assigner / odrl:assignee

### LINKED ISSUE #195: Change `dspace:providerId` to use `assigner` instead - and `dspace:consumerId` to `assignee`
As described in ODRL
https://www.w3.org/TR/odrl-model/#policy-offer
an `Offer` MUST have one `assigner`. Which is basically what has been intended with `dspace:providerId`.

An `Agreement` MUST have both, `assigner` AND `asignee`

As described for compact policies:
https://www.w3.org/TR/odrl-model/#composition-compact
these properties can be set on the Policy level and being applied to all its rules.

**In the catalog** we will use the same mechanism as we do with the `target` (#192 ). The `assigner` needs to be coming from the enclosing context. Here, not the `dataset` but the `Catalog`.
This requires #196  to introduce a `participantId` on `Catalog` level!

In `ContractRequestMessage->dspace:offer` the `assigner` needs to be set (on the Offer/Policy level!) (The value is a `participantId`)
(We agreed to alwys use 'compact Policy' form and we need to 'profile' this. TODO)

Similar for `assignee`. This is a MUST in an `Agreement` and will replace `dspace:consumerId`.

Beware, that `assigner` and `asignee` are of type `Party` and require to be an IRI. Pure literals are not allowed.

Any further comments @jimmarino @sebbader-sap ?

CC: @juliapampus @ssteinbuss 

--
Matthias Binzer

### PR DESCRIPTION
**Pull request by sebbader-sap**
Closes #195

### CODE REVIEW

File: negotiation/contract.negotiation.binding.https.md
Comment (juliapampus) [inline negotiation/contract.negotiation.binding.https.md:1]: This ToC is a real nighmare (especially rendered). Can you please remove the level 3+ headers for sec 2 and 3 from it (i.e. GET/POST, request, response)

**Review by juliapampus — CHANGES_REQUESTED** [review]
"Solves" is not a [closing keyword](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) for GitHub. Please use one of them, this automates closing issues when a PR is merged.

You only checked for `dspace:providerId` and `dspace:consumerId`. Please also search for `providerId` and `consumerId`. For instance, the protocol document (line 102) contains some leftovers.

**Review by mkollenstart — APPROVED** [review]
Since it is not really part of this pull request, I'm still approving, but I think we should set a bit more representative identifiers in the examples. Especially in the case of URNs.

File: negotiation/contract.negotiation.binding.https.md
Comment (sebbader-sap) [reply negotiation/contract.negotiation.binding.https.md:1]: done

**Comment by sebbader-sap** [comment]
> You only checked for `dspace:providerId` and `dspace:consumerId`. Please also search for `providerId` and `consumerId`. For instance, the protocol document (line 102) contains some leftovers.

@juliapampus I have pushed the changes, please have a quick look.

**Review by ssteinbuss — APPROVED** [review]

**Review by juliapampus — APPROVED** [review]

**Pull request merged**